src/Illuminate/Queue/Console/RetryCommand.php

```diff
@@ -54,6 +54,12 @@ protected function getJobIds()
 
         if (count($ids) === 1 && $ids[0] === 'all') {
             $ids = Arr::pluck($this->laravel['queue.failer']->all(), 'id');
+        } elseif (count($ids) === 1 && count($rangeEdges = explode('-', $ids[0])) == 2 && $rangeEdges[1] > 0) {
```

Comment:
What about using something like this instead:

    } elseif (count($ids) == 1 && sscanf($ids[0], '%d-%d', $rangeMin, $rangeMax) == 2 && $rangeMin > 0 && $rangeMax > $rangeMin) {
        $ids = range($rangeMin, $rangeMax);
    }

This would not only simplify the code but also be more stricter. 👍

Reply:
I think it is better to abstract out `count($ids) === 1` into an surrounding `if` check because it is common to both branches

Reply:
Another alternative that won't care about the order of the range, ie 4-5 or 5-4 would be:

```
} elseif (count($ids) === 1 && count($range = explode('-', $ids[0])) == 2) {
  $ids = range(min($range), max($range));
}
```

@Dylan-DPC No need to add complexity of nesting if statements my trying to remove the duplication of `count($ids) === 1`. Duplication is cheap and in this case is a whole lot easier to read.

Reply:
It doesn't add any complexity. And contrary to what you say, nested if statements make it easier to read. It is easier to note what happens if count is 1 and when count isn't 1 without bothering about the other conditions.